From: SeongJae Park <sj@kernel.org>
Date: Tue, 23 Dec 2025 17:01:15 -0800
Subject: [PATCH] ==== sysfs cleanup dirs undeer failures ====

mm/damon/sysfs: free setup failures generated zombie sub-sub dirs

Some DAMON sysfs directory setup functions generates its sub and sub-sub
directories.  For example, 'monitoring_attrs/' directory setup creates
'intervals/' and 'intervals/intervals_goal/' directories under
'monitoring_attrs/' directory.  When such sub-sub directories are
successfully made but followup setup is failed, the setup function
should recursively clean up the subdirectories.

However, such setup functions are only dereferencing sub directory
reference counters.  As a result, under certain setup failures, the
sub-sub directories keep having non-zero reference counters.   It means
the directories cannot be removed like zombies, and the memory for the
directories cannot be freed.

The user impact of this issue is limited due to the following reasons.

When the issue happens, the zombie directories are still taking the
path.  Hence attempts to generate the directories again will fail,
without additional memory leak.  This means the upper bound memory leak
is limited.  Nonetheless this also implies controlling DAMON with a
feature that requires the setup-failed sysfs files will be impossible
until the system reboots.

Also, the setup operations are quite simple.  The certain failures would
hence only rarely happen, and are difficult to artificially trigger.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_m8cgeycy | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_m8cgeycy

diff --git a/hkml_cv_bogus/hkml_cv_bogus_m8cgeycy b/hkml_cv_bogus/hkml_cv_bogus_m8cgeycy
new file mode 100644
index 000000000000..e69de29bb2d1
-- 
2.47.3

